solawi/api.py

```diff
@@ -148,51 +167,68 @@ def shares_details(share_id):
 
 @api.route("/shares/<int:share_id>/deposits", methods=["GET"])
 @jwt_required()
-def share_deposits(share_id):
+def share_deposits(share_id: int):
     deposits = Share.get_deposits(share_id)
     return jsonify(deposits=deposits)
 
 
 @api.route("/shares/<int:share_id>/bets", methods=["GET"])
 @jwt_required()
-def share_bets(share_id):
+def share_bets(share_id: int):
     bets = Share.get_bets(share_id)
     return jsonify(bets=bets)
 
 
-@api.route("/shares/<int:share_id>/bets", methods=["POST", "PUT"])
+class BetSchema(BaseModel, extra=Extra.forbid):
+    value: int
+    start_date: date
+    end_date: Optional[date]
+
+
+@api.route("/shares/<int:share_id>/bets", methods=["POST"])
```

Comment:
This is just splitting one route definition with `POST` and `PUT` into two to make it a bit easier to skim